src/app/space/create/deployments/deployments-donut/deployments-donut-chart/deployments-donut-chart.component.ts

```diff
@@ -0,0 +1,245 @@
+import { Component, DoCheck, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core';
+
+import { debounce, get, isEmpty, isEqual, reject, size, uniqueId } from 'lodash';
+
+import * as c3 from 'c3';
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'deployments-donut-chart',
+  templateUrl: './deployments-donut-chart.component.html',
+  styleUrls: ['./deployments-donut-chart.component.less']
+})
+export class DeploymentsDonutChartComponent implements DoCheck, OnInit, OnChanges, OnDestroy {
+
+  @Input() pods: any;
+  @Input() mini: boolean;
+  @Input() desiredReplicas: number;
+  @Input() idled: boolean;
+
+  chartId = uniqueId('deployments-donut-chart');
+  podStatusData: any;
+  total: number;
+
+  private phases = ['Running',
+    'Not Ready',
+    'Warning',
+    'Error',
+    'Pulling',
+    'Pending',
+    'Succeeded',
+    'Terminating',
+    'Unknown'];
+
```

Comment:
I think the preferred style for multiline array literals is with the brackets on separate lines from the first and last elements, but not a big deal.